client/clientfactory.go

```diff
@@ -34,26 +34,26 @@ type Factory interface {
 	NewMatchingClient() (matching.Client, error)
 }
 
-type tchannelClientFactory struct {
-	ch                    *tchannel.Channel
+type dispatcherClientFactory struct {
```

Comment:
how about renaming this to rpcClientFactory ? Dispatcher is just an implementation detail specific to yarpc

Reply:
Done.